src/main/java/dev/lu15/voicechat/network/voice/GroupManager.java

```diff
@@ -0,0 +1,33 @@
+package dev.lu15.voicechat.network.voice;
+
+import dev.lu15.voicechat.network.minecraft.Group;
+import net.minestom.server.entity.Player;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+
+public class GroupManager {
```

Comment:
This class should have methods to create, destroy and modify groups, which will take the group management code out of the packet processing classes. This'll also allow for events to be created in order to increase the customisation of the implementation.